### PR TITLE
Connect store to components

### PR DESCRIPTION
**Pull request by NatiDeme**
# Changes in this PR:
- Data rendered from the store.
- Imported Provider into the main document and connected redux to the component.
- I have used useSelector, and useDispatch hooks to access actions and states from the store.
- No linter errors

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,31 +1,42 @@
...
 
+const bookReducer = (state = allBooks, action) => {
+  switch (action.type) {
+    case ADD_BOOK:
+      return {
+        books: [...state.books, action.payload],
+      };
+    case REMOVE_BOOK:
+      return {
+        books: [...state.books.filter((i) => i.title !== action.book.title)],
```
Comment (TedLivist) [inline src/redux/books/books.js]: - [x] With this setup books of the same title will be filtered out of the state, but it should not be so. Your books should be filtered using the `id`. Kindly do that.

File: src/components/Book.js [hunk 2]
```diff
@@ -12,7 +19,7 @@ function Book(props) {
             <p className="book-title">{i.genre}</p>
             <p className="book-title">{i.title}</p>
             <p className="book-author">{i.author}</p>
-            <button type="button">Remove</button>
+            <button type="button" onClick={() => removeNow(i)}>Remove</button>
```
Comment (TedLivist) [inline src/components/Book.js:22]: - [ ] **[OPTIONAL]** While implementing this mapping, you are using the entire book object as the `key` `(<div key={i} className="name-wrap">)`. I would recommend you use something unique like the book `id` property to clear this warning below:

  unique-id

File: src/components/AddBook.js [hunk 3]
```diff
@@ -1,14 +1,36 @@
...
+  const dispatch = useDispatch();
+  const newbook = {
+    id: '',
+    title: '',
+    author: '',
+  };
+  const [mybook, setMyBook] = useState(newbook);
+  const addNew = () => {
+    dispatch(addBook(mybook));
+  };
```
Comment (TedLivist) [inline src/components/AddBook.js]: - [x] **[OPTIONAL]** I think it would be nice to clear the inputs after dispatching the book.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @NatiDeme,

Good job so far!
✔️ Good documentation
✔️ Inputs are set by local state
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the **[OPTIONAL]** prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Gambit142 — APPROVED** [review]
Hi @NatiDeme 👋, 

## Status: Approved 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Great work implementing all the changes suggested by the previous reviewer. 👍 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**